Sources/SafeDICore/Generators/DependencyTreeGenerator.swift

```diff
@@ -20,7 +20,7 @@
 
 import Collections
 
-public final class DependencyTreeGenerator {
+public actor DependencyTreeGenerator {
```

Comment:
We were getting an error that access to `DependencyTreeGenerator` was racy. Turns out all the API here is already async, so might as well make this type an `actor`